app/bin/server.dart

```diff
@@ -1,121 +1,16 @@
 
 void main() {
-  useLoggingPackageAdaptor();
-
-  withAppEngineServices(() async {
-    return withCorrectDatastore(() async {
-      final shelf.Handler apiHandler = await setupServices(activeConfiguration);
-
-      await runAppEngine((HttpRequest ioRequest) async {
-        return withCorrectDatastore(() {
-          if (context.isProductionEnvironment &&
-              ioRequest.requestedUri.scheme != 'https') {
-            final secureUri = ioRequest.requestedUri.replace(scheme: 'https');
-            ioRequest.response
-              ..redirect(secureUri)
-              ..close();
-          } else {
-            try {
-              return shelf_io.handleRequest(ioRequest,
-                  (shelf.Request request) async {
-                logger.info('Handling request: ${request.requestedUri}');
-                await registerLoggedInUserIfPossible(request);
-                try {
-                  final sanitizedRequest = sanitizeRequestedUri(request);
-                  return await appHandler(sanitizedRequest, apiHandler);
-                } catch (error, s) {
-                  logger.severe('Request handler failed', error, s);
-                  return new shelf.Response.internalServerError();
-                } finally {
-                  logger.info('Request handler done.');
-                }
-              });
-            } catch (error, stack) {
-              logger.severe('Request handler failed', error, stack);
-            }
-          }
-        });
-      });
-    });
-  });
-}
-
-Future<shelf.Handler> setupServices(Configuration configuration) async {
-  registerTemplateService(
-      new TemplateService(templateDirectory: TemplateLocation));
-
-  final bucket = storageService.bucket(configuration.packageBucketName);
-  final tarballStorage = new TarballStorage(storageService, bucket, null);
-  registerTarballStorage(tarballStorage);
-
-  initOAuth2Service();
-
-  await initSearchService();
-
-  final cache = new AppEnginePackageMemcache(memcacheService, '');
-  initBackend(cache: cache);
-
-  UploadSignerService uploadSigner;
-  if (configuration.hasCredentials) {
-    final credentials = configuration.credentials;
-    uploadSigner = new ServiceAccountBasedUploadSigner(credentials);
-  } else {
-    final authClient = await auth.clientViaMetadataServer();
-    registerScopeExitCallback(authClient.close);
-    final email = await obtainServiceAccountEmail();
-    uploadSigner =
-        new IamBasedUploadSigner(configuration.projectId, email, authClient);
-  }
-  registerUploadSigner(uploadSigner);
-
-  return new ShelfPubServer(backend.repository, cache: cache).requestHandler;
-}
-
-shelf.Request sanitizeRequestedUri(shelf.Request request) {
-  final uri = request.requestedUri;
-  final resource = uri.path;
-  final normalizedResource = path.normalize(resource);
-
-  if (resource == normalizedResource) {
-    return request;
+  if (Platform.environment['PUB_SERVICE'] == 'analyzer') {
+    analyzer.main();
```

Comment:
Please use the 'GAE_SERVICE' environment variable instead, then we can also have 3 cases: the analyzer service, the default service and an error if this script is invoked with any other service.